src/main/java/com/thebois/models/beings/Colony.java

```diff
@@ -172,4 +192,32 @@ public int numberOf(final ItemType itemType) {
         return inventory.numberOf(itemType);
     }
 
+    /**
+     * Listens to the spawnPawnsEvent in order to know when to add pawns to the colony.
+     *
+     * @param event The published event.
+     */
+    @Subscribe
+    public void onSpawnPawnsEvent(final SpawnPawnsEvent event) {
+        // final Iterable<Position> vacantPositions = positionFinder.findEmptyPositions(10);
+        final Iterable<Position> vacantPositions =
+            positionFinder.tryGetEmptyPositionsNextTo(event.getSpawnPosition(),
+                                                      event.getNumberOfPawns(),
+                                                      event.getSpawnRadius());
+        final Random random = new Random();
+        for (final Position vacantPosition : vacantPositions) {
+            addBeing(new Pawn(vacantPosition, vacantPosition, random, pathFinder, structureFinder));
+        }
+    }
+
+    @Serial
+    private void readObject(final java.io.ObjectInputStream in) throws IOException,
+                                                                       ClassNotFoundException {
+        // Registers every time on deserialization because it might be registered to an old instance
+        // of the event bus.
+        // (caused by saving/loading).
+        Pawntastic.getEventBus().register(this);
+        in.defaultReadObject();
+    }
```

Comment:
Do we really need this? The only reason we have it in abstractbeing was because they didn't immediately start listening to the events but finished their path then started listening to the correct one. How and why? I don't know. But here we should run into this problem since colony wont be occupied with something before right?

Reply:
And it doesn't do that _at all_? What we noticed with `Pawn`s was that they were busy with walking their path and didn't listen to the events until they finished walking that path, then they started to listen to the events again. But since colony wont be busy with a task that takes a while it shouldn't be a problem?

Reply:
Ok I'll take a look if it is not needed